strawberryfields/decompositions.py

```diff
@@ -77,6 +77,40 @@ def takagi(N, tol=13):
     return rl, U
 
 
+def graph_embed(mat, max_mean_photon=1.0, make_traceless=True, tol=6):
+    r""" Given an symmetric adjacency matrix (that can be complex in general),
+    it returns the squeezing parameter and interferometer necessary for
+    implementing it in GBS.
+
+    Args:
+        mat (array): square symmetric complex (or real or integer) array
+        max_mean_photon (float): threshold value. It guarantees that the mode with
+            the largest squeezing has ``max_mean_photon`` as the mean photon number
+            i.e. :math:`sinh(r_{max})^2 == max_mean_photon`
+        make_traceless (boolean): removes the trace of the input matrix.
```

Comment:
Note: when this is clarified, the same thing appears in strawberryfields/ops.py

Reply:
```suggestion
        make_traceless (boolean): removes the trace of the input matrix, by performing the transformation
            :math:`\tilde{A} = A-\mathrm{tr}(A) \I/n`. This may reduce the amount of squeezing needed to encode
            the graph.
```

Reply:
So the matrix `mat` is the one that is made traceless? 
If so, it might confuse users that we earlier call it an `adjacency matrix` (usually this means the diagonal entries are already zero)